my-webxr-app/src/utils/PcaCovariance.tsx

```diff
@@ -70,29 +69,67 @@ export function computeEigenvectorsFromCovarianceMatrix(covarianceMatrix: Matrix
 }
 
 /**
- * This function performs PCA on a data set using covariance method.
+ * Calculates the variance explained by each principal component, in descending order.
+ *
+ * @param {Matrix} covarianceMatrix - The covariance matrix to compute the variance explained.
+ * @returns {number[]} An array of variances explained by each principal component.
+ * @throws {Error} If an error occurs during the PCA computation.
+ */
+export function calculateVarianceExplained(covarianceMatrix: Matrix): number[] {
+  try {
+    const eigenvalues = computeEigenvaluesFromCovarianceMatrix(covarianceMatrix);
+    const totalVariance = eigenvalues.reduce((a, b) => a + b, 0);
+    return eigenvalues.map((value) => value / totalVariance);
+  } catch (e) {
+    // may need to surface this error somewhere
+    // console.error(`An error occurred during calculating variance explained: ${e}`);
+    return [];
```

Comment:
Is this a fatal error? As in one that means conducting PCA failed? If so, we will probably have to inform the user if such a case occurs. Would throwing an Exception be reasonable? + logging once we get that implemented.

Reply:
Yes, something like a null data point get past the filtering before the PCA.

Reply:
We could catch the errors and rethrown them here. As right now I just don't want this to break on the user end and I am not sure how we going to notify user yet.